### DISCUSSION TITLE
Persist Bond Stereo on Reading In

### DISCUSSION THREAD

**Discussion by huwdjones** [rdkit/rdkit#4409] [Q&A] [answered]
Apologies if this has been asked several times before but is it possible to persist the bond stereo assignments on reading in a molfile? The reason I ask is that I have a third party chemical sketch package and I am trying to generate an RDKit SVG image from the sketch drawn. I find that in some cases the bond stereo can change on reading in the molfile generated from the sketch package with Chem.MolFromMolBlock(). In the example below the compound is drawn with the solid wedge on the C-Cl bond however this is then transformed to the dashed wedge on the C-C bond on reading in.

```
import difflib
from rdkit.Chem import MolFromMolBlock, MolToMolBlock

molstring_from_sketch = '''[(S)-1-Chloro-1-fluoroethyl]benzene with wedge C-Cl
                      

 10 10  0  0  0  0  0  0  0  0999 V2000
   21.3079  -15.5891    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   22.0223  -16.0016    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   22.0223  -16.8266    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   21.3079  -17.2391    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   20.5935  -16.8266    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   20.5935  -16.0016    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   21.3079  -14.7641    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   20.5935  -14.3516    0.0000 F   0  0  0  0  0  0  0  0  0  0  0  0
   22.0223  -14.3516    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   21.3079  -13.9391    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  2  0
  3  4  1  0
  4  5  2  0
  5  6  1  0
  6  1  2  0
  1  7  1  0
  7  8  1  0
  7  9  1  0
  7 10  1  1
M  END
'''

mol = MolFromMolBlock(molstring_from_sketch, sanitize=False)
print(*iter(difflib.context_diff(molstring_from_sketch, MolToMolBlock(mol))))
```

I have seen some older threads that suggest preprocessing of the molfile to record the original bond stereo and then reapply this but I wondered whether there was anything else available in the toolkit to meet this requirement?

Many thanks as ever.

Huw Jones

**Comment by greglandrum** [comment, marked as answer]
When reading a mol file the RDKit parses the wedging information which is there and translates it to its own internal representation. When it needs to wedge bonds again - for example when generating an image - it determines where to place wedges based on an RDKit algorithm.
If you want to re-apply the same wedging information which was in the input mol block, you'll need to do so manually. The RDKit does preserve info about the original wedging, so you don't have to deal with parsing the file yourself. Here's an example function for this:
https://github.com/chembl/ChEMBL_Structure_Pipeline/blob/master/chembl_structure_pipeline/standardizer.py#L469
Note that function only works for V2000 mol blocks; the property for V3000 mol blocks is called `_MolFileBondCfg` and the codes are different. The mapping is:
| `_MolFileBondCfg` | `_MolFileBondStereo` |
|-------|-------|
| 1 | 1 |
| 2 | 3 |
| 3 | 6 |

**Reply by huwdjones** [reply]
Thanks Greg that's great!
